### ISSUE TITLE
optimize keyword argument for tf.einsum

### ISSUE THREAD

**Issue by roymiles** [tensorflow/tensorflow#38210]
I am trying to use the ```optimize``` keyword argument for the einsum operation, but I receive the following error:

```
    ....
    my_path/file.py:27 basis_transform_conv2d  *
        out = tf.einsum('bhwcfg,fgcu->bhwfgu', x_col, basis, optimize='optimal')
    Anaconda3\envs\tf\lib\site-packages\tensorflow_core\python\ops\special_math_ops.py:252 einsum
        [format(key) for key in sorted(list(kwargs.keys()))]))

    TypeError: invalid keyword arguments for this function: optimize
```

I am currently using tensorflow-gpu==2.0.0 with up-to-date CUDA, cudnn (everything else works fine).
It is to my understanding that tensorflow is calling the legacy einsum operation, einsum_v1 : https://github.com/tensorflow/tensorflow/blob/44d3065391ea2188979ff2e747f4e7b789edd2e6/tensorflow/python/ops/special_math_ops.py#L407

How do I ensure my code calls einsum_v2, which supports the ```optimize``` keyword argument?

**Comment by gadagashwini** [comment]
@roymiles, Can you provide the minimal code snippet to replicate the reported issue. Thanks!

**Comment by roymiles** [comment]
`
import tensorflow as tf

a = tf.zeros(shape=(100, 100))
b = tf.zeros(shape=(100, 300))

c = tf.einsum('ij,jk->ik', a, b, optimize=True)
`

Is enough to reproduce the error. In the documentation, **kwargs should allow an `optimize` keyword argument, similar to np.einsum.

In case anyone is having the same problem, https://optimized-einsum.readthedocs.io/en/stable/ is a good solution that supports tensorflow GPU backend.

**Comment by gadagashwini** [comment]
I could reproduce the issue with Tf 2.0.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/5468d5657551c0061b482b500c99bc1a/untitled493.ipynb). Thanks!

**Comment by gadagashwini** [comment]
@roymiles,
This issue is fixed in Tf 2.2.rc1
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/8da934854e3a5f83e9311f9ef740788e/untitled494.ipynb). Thanks!

**Comment by ravikyram** [comment]
@roymiles 
This issue was resolved in  TF version 2.2.rc1 .Please verify once and close the issue. Thanks!
